### PR TITLE
Update openbare

### LINKED ISSUE #25: Hard coded URL
The Web Console URL is [hard coded](https://github.com/openbare/openbare/blob/cc7eeec63766757f9f5d370be1deedb0193ee98b/library/amazon_account_utils.py#L162) to suse-demo.

Switch to using a settings option so this can be set per deployment.

### LINKED ISSUE #26: Unicode in name causes error with AWS
If a user has authenticated to openbare with a username that includes unicode characters and attempts to checkout an AWS lendable, boto3 raises a validation error.

AWS IAM usernames must be alphanumeric including +=,.@_- (case insensitive) and less than 65 characters.

Currently another bug exists as well if the authentication backend is case sensitive like Django. E.g. if two users authenticate with the usernames John and john only one could have an AWS lendable checked out.

### LINKED ISSUE #27: Renewal limit exceeded raises attribute error
If a user manages to trigger the renewal URL after the lendable has been renewed the max amount of times a ValidationError is raised. An unhandled exception is raised while handling the Validation Error.

>During handling of the above exception (['No more renewals are available for this item.']), another exception occurred:
> 'ValidationError' object has no attribute 'error_dict'

https://github.com/openbare/openbare/blob/master/library/views.py#L87

### PR DESCRIPTION
**Pull request by smarlowucf**
Add unit tests for the library app, cleanup flake8 warnings, update readme and account for unicode in usernames for AWS.

This PR closes #25, resolves #26 and fixes #27.

Additional changes include:
- Use login_required decorator instead of custom function.
- Remove admin_emails from context as it is not used in the template.
- Update Django version requirement.
- Add requirements files for use in setting up dev environment.
- Cleanup deprecation warnings.
- Cleanup checkout, use getattr method to access AWS_IAM_GROUP settings value with a default of None.
- Use class based views for index and checkout to remove some code duplication.

New methods _set_username and _validate_username have been added to the Lendable class. Proxy models can override these methods to provide lendable specific username validation. If a username does not pass validation or username is in use, _set_username will auto-generate a new username for checkout.

This will allow two users with usernames that normalize to the same value the opportunity to checkout a given lendable. E.g. John and Jőhn would both normalize to John for AWS and would collide. Thus, preventing one user from checking out an IAM account. Because Django is case sensitive and AWS is not, this would also be possible with two users john and John.

With the requirement update to ~1.10 the code is now using is_anonymous as an attr instead of a method. This means the changes will be incompatible with previous versions.

### CODE REVIEW

File: library/views.py [hunk 1]
```diff
@@ -177,6 +175,9 @@ def get_items_checked_out_by(user=None):
 
 
 def _verify_user(user=None):
+    warnings.warn('_verify_user is deprecated and will be removed.'
```
Comment (bear454) [inline library/views.py]: I wouldn't bother with this; just remove it since it's unused.

File: library/tests.py [hunk 2]
```diff
@@ -15,6 +17,115 @@
 # You should have received a copy of the GNU General Public License
 # along with openbare. If not, see <http://www.gnu.org/licenses/>.
 
-from django.test import TestCase
```
Comment (bear454) [inline library/tests.py]: ?

File: library/tests.py [hunk 3]
```diff
@@ -60,9 +67,9 @@ def test_lendable_flow(self):
         self.c.login(username=self.user.username, password='str0ngpa$$w0rd')
 
         # Test check out lendable
-        response = self.c.post(reverse('library:checkout',
-                                       args=['lendable']),
-                               follow=True)
+        response = self.c.get(reverse('library:checkout',
```
Comment (bear454) [inline library/tests.py]: Why `get` ? These actions _should_ map to `HTTP POST`.

File: library/amazon_account_utils.py [hunk 4]
```diff
@@ -171,9 +174,15 @@ def create_iam_account(self, username, group=None):
             The required credentials.
         """
         self.log("creating IAM user '%s'" % username, INFO)
+
+        alias = getattr(settings, 'AWS_ACCOUNT_ID_ALIAS', None)
```
Comment (bear454) [inline library/amazon_account_utils.py]: :+1: 

Please add the attribute to the settings template as well.

File: README.md [hunk 5]
```diff
@@ -20,7 +20,7 @@ Dependencies
 ------------
 
 * python > 3.2
-* Django ~ 1.8.4
+* Django >= 1.10
```
Comment (bear454) [inline README.md]: Let's try to keep the version consistent here and our spec, which is 

```
Requires:       python3-Django = 1.10.2
```

**Comment by bear454** [comment]
@smarlowucf looks great - just a couple of questions in comments.

File: README.md (hunk 5 above)
Comment (smarlowucf) [inline README.md]: Sounds good. :+1: 

Is there any way in the spec to list the version in a compatible way? So any compatible version updates are automagically downloaded?

Like: ~= 1.10 or >= 1.10 or == 1.10.*

File: library/views.py (hunk 1 above)
Comment (smarlowucf) [inline library/views.py]: :+1:

File: library/amazon_account_utils.py (hunk 4 above)
Comment (smarlowucf) [inline library/amazon_account_utils.py]: I think they should be good to go but I'll double check.

File: library/tests.py (hunk 3 above)
Comment (smarlowucf) [inline library/tests.py]: Right now they are all GET requests because they are HTML links, except the extension request which is a form POST. However, the function based views would accept GET or POST. Class based views are explicitly set. For example I used the class based get method to process the request so only get works. I could move the code to dispatch or add a post method which would work similar to the function based views.

File: library/tests.py (hunk 2 above)
Comment (smarlowucf) [inline library/tests.py]: That's still there just moved to the end of [line 23](https://github.com/openbare/openbare/pull/28/files/1c8db2437abc7aded37caf08b5eba3e525348769#diff-efc6577027adc384126b4470b21d4331R23)

File: README.md (hunk 5 above)
Comment (bear454) [inline README.md]: See 2c9a060 ; PR #29.

Comment (smarlowucf) [inline README.md]: Great :+1:. That way we can always pull in the non-breaking security updates without requiring changes to spec or requirements.

**Comment by rjschwei** [comment]
Looks pretty good overall. I think it could have been broken up into multiple smaller requests.

**Comment by smarlowucf** [comment]
@rjschwei agreed.
